### PR TITLE
Bug 1212865 - Add and improve ordering of related instances

### PR DESCRIPTION
**Pull request by jwhitlock**
For API-backed tables with multiple elements, such as [Web/CSS/background](https://developer.mozilla.org/en-US/docs/Web/CSS/background#Browser_compatibility), the order of the table rows is usually different than the source MDN table. This PR implements the features and fixes the bugs so the [importer](https://browsercompat.herokuapp.com/importer/651) will save the order of feature and specification rows ([bug 1212865](https://bugzilla.mozilla.org/show_bug.cgi?id=1212865))

The changes are:
* The importer sets the row orders by specifying ``feature.children`` (order for browser compatibility table) and ``feature.sections`` (order for specification table)
* The API will set the order of related elements on PUT/POST/PATCH, including the other two ordered relations, ``browser.versions`` and ``specification.sections``
* The external tools will set the order of related elements on bulk import
* The API will remember the order in the historical instance
* The instance cache will serialize / deserialize the IDs of related elements in order
* The API will return related items in the order
* The sample JS display code will display rows in that order

Getting all this working required fixing some additional bugs:
* [bug 1159337](https://bugzilla.mozilla.org/show_bug.cgi?id=1159336) - Use the order-aware ``TreeManager`` for features, so that ``feature.children`` are returned in [MPTT](https://github.com/django-mptt/django-mptt) order rather than random MySQL order.
* [bug 1159317](https://bugzilla.mozilla.org/show_bug.cgi?id=1159317) - Fix a bug when doing a PUT to update a view_feature with incomplete data, used by the importer when re-importing a page.
* [bug 1159292](https://bugzilla.mozilla.org/show_bug.cgi?id=1159292) - Attempt to restore related item order when reverting to a past version of an instance.
* A newly discovered bug about instance cache invalidation for common cases such as creating a new instance or closing a changeset.

These changes had a large impact on the documentation / integration tests (specified in docs/doc_cases.json, with results in docs/raw/), which can mostly be ignored outside of running ``make test-integration`` to validate them.

### CODE REVIEW

**Comment by jwhitlock** [comment]
Rebased on master

File: webplatformcompat/models.py [hunk 1]
```diff
@@ -349,7 +416,8 @@ def post_save_update_cache(sender, instance, created, raw, **kwargs):
         delay_cache = getattr(instance, '_delay_cache', False)
         if not delay_cache:
             from .tasks import update_cache_for_instance
-            update_cache_for_instance(name, instance.pk, instance, False)
+            update_cache_for_instance(
+                name, instance.pk, instance, update_only=False)
```
Comment (groovecoder) [inline webplatformcompat/models.py]: Seems to call `update_cache_for_instance` with `update_only=False` a lot ... should we change the default parameter value to `False`?

File: webplatformcompat/models.py [hunk 2]
```diff
@@ -85,12 +96,39 @@ class Feature(MPTTModel):
...
+
+        # Set order, refreshing as we go
+        prev_child = None
+        moved = False
+        for pos, next_child in enumerate(children):
+            if current_children[pos].pk != next_child.pk:
+                if moved:
+                    next_child.refresh_from_db()
+                if prev_child is None:
+                    next_child.move_to(self)
```
Comment (groovecoder) [inline webplatformcompat/models.py]: Maybe include the explicit argument value for `move_to`

Comment (jwhitlock) [inline webplatformcompat/models.py]: Updated to ``first-child``, which is good, because the default was ``last-child``.

**Comment by groovecoder** [comment]
Updated, but I don't see a new commit?

**Comment by jwhitlock** [comment]
Rebased, and made some changes:
* Used ``next_child.move_to(self, "first-child")`` when placing the first child feature in order
* Changed default to ``update_cache_for_instance(..., update_only=False)``

File: webplatformcompat/tests/test_tasks.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def test_update_cache_no_invalidation(self):
         self.mock_cache.update_instance.return_value = []
         update_cache_for_instance('Maturity', self.mat.id)
         self.mock_cache.update_instance.assert_called_once_with(
-            'Maturity', self.mat.id, None, None, update_only=True)
+            'Maturity', self.mat.id, None, None, update_only=False)
```
Comment (groovecoder) [inline webplatformcompat/tests/test_tasks.py]: Still being explicit with these `update_only` argument values even though the parameter default is `False`?

File: webplatformcompat/tasks.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 @shared_task(ignore_result=True)
 def update_cache_for_instance(
         model_name, instance_pk, instance=None, version=None,
-        update_only=True):
+        update_only=False):
     cache = Cache()
     invalid = cache.update_instance(
         model_name, instance_pk, instance, version, update_only=update_only)
```
Comment (jwhitlock) [inline webplatformcompat/tasks.py]: This is what is being mocked in test_tasks.py - explicit arguments and all

File: webplatformcompat/tests/test_tasks.py (hunk 3 above)
Comment (jwhitlock) [inline webplatformcompat/tests/test_tasks.py]: The mock is for ``cache.update_instance``, which is being called with explicit parameters - see webplatformcompat/tasks.py

Comment (groovecoder) [inline webplatformcompat/tests/test_tasks.py]: Duh. Sorry I didn't notice this was a mock.

**Comment by groovecoder** [comment]
Cool, looks good. Good update.

**Pull request merged**